teps/0007-conditions-beta.md

```diff
@@ -0,0 +1,612 @@
+
+This design will give us the flexibility to handle both simple `Guards` (e.g. checking a branch name) using string expressions and complex `Guards` (e.g. those that need a `Workspace` and/or `Volume` with the files to execute on) in a `Pod`, while providing more options when a `Guard` evaluates to `False`. 
+
+
+### Simplicity
+
+As discussed in the background section, `Conditions` manifest themselves as `Tasks`. We want to keep Tekton as simple as possible by reusing existing components. So we propose phasing out the separate `Conditions` CRD and eventually deprecating it. In place of `Conditions`, we want to use `Tasks` - referred to as `Guards` - to determine whether or not a `Task` executes. We refer to them as `Guards` because they determine if a `Task` executes, not if/else as would be expected from a `Condition`; more details on `Guards` vs `Conditions` naming can be found in [this issue](https://github.com/tektoncd/pipeline/issues/2635). Thus, we won’t have `Conditions` to migrate to beta and won’t have to maintain the separate `Conditions` CRD. Moreover, the features supported by `Tasks` become readily available to be used for guarded execution of `Tasks`. 
+
+A `Task` that is used as a `Guard` will produce a `Result` called `Skip` with values `True` or `False` -- to indicate whether the `Task` it’s guarding will be executed or not. When we have more than one `Guard`, the `Task` will be executed when all the `Guards` evaluate to `True`. Using `Results` allows us to distinguish between when a `Task` fails and when the `Guard` evaluates to `False`. 
+
+In the example of checking whether a file exists, the `Task` that would replace the `Condition` would be specified in a list of `when-expressions` as such:
+
+```
+apiVersion: tekton.dev/v1alpha1
+kind: Condition
+metadata:
+  name: file-exists
+spec:
+  parameters:
+    - name: "path"
+  resources:
+    - name: workspace
+      type: git
+  check:
+    image: alpine
+    script: 'test -f $(resources.workspace.path)/$(params.path)'
+---
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: file-exists
+spec:
+  parameters:
+    - name: "path"
+  resources:
+    - name: workspace
+      type: git
+  results:
+    - name: skip
+      description: boolean indicating whether guarded Task should be skipped
+  steps:
+  - name: check-file-exists
+    image: alpine
+    script: |
+      if [ -f $(resources.workspace.path)/$(params.path) ]; then
+        echo true | tee /tekton/results/skip
+      else
+        echo false | tee /tekton/results/skip
+      fi
+---
+api: tekton.dev/v1alpha1
+kind: Pipeline
+metadata: 
+    name: generate-file
+    spec: 
+      tasks: 
+        - name: create-file
+          taskRef: 
+            name: create-readme-file 
+        - name: check-file-exists 
+          when:
+            - name: file-exists
+              taskRef: 
+                name: file-exists
```

Comment:
we also need to pass in workspaces here... I also forgot that in my example

Reply:
added them, thanks :)